packages/core/src/components/search/Filter/FilterSlider.tsx

```diff
@@ -166,13 +166,32 @@ function FilterSlider({
         }
       : undefined
 
+  const allDeliveryMethodsFacetValue = {
+    value: 'all-delivery-methods',
+    label:
+      deliverySettingsData?.deliveryMethods?.allDeliveryMethods ??
+      'All delivery methods',
+    selected: !selected.some(
+      (facet) =>
+        facet.key === 'shipping' && facet.value !== 'all-delivery-methods'
+    ),
+    quantity: 0,
+  }
+
   let filteredFacets = facets.filter((facet) => facet.key !== 'shipping')
   if (isDeliveryPromiseEnabled) {
     filteredFacets = facets.map((facet) => {
       if (
         facet.key === 'shipping' &&
         facet.__typename === 'StoreFacetBoolean'
       ) {
+        const hasAllDeliveryMethodsFacet = facet.values.some(
+          (item) => item.value === allDeliveryMethodsFacetValue.value
+        )
+        if (!hasAllDeliveryMethodsFacet) {
+          facet.values = [allDeliveryMethodsFacetValue, ...facet.values]
```

Comment:
If you agree with the suggestion.

```suggestion
          facet.values = [allDeliveryMethodsFacet, ...facet.values]
```